app/models/chat_room.rb

```diff
@@ -0,0 +1,4 @@
+class ChatRoom < ApplicationRecord
+  belongs_to :race
+  has_many :chat_messages, dependent: :destroy
+end
```

Comment:
What's the purpose of having this be its own model, rather than setting `belongs_to :race` on `ChatMessage`?

Reply:
I use chat room to hold a locked at time stamp so that you can only send messages upwards of X minutes after the end of a race.  Is that behavior desirable or something not really needed?

Reply:
I like that idea -- can we enforce it through logic rather than data? i.e. if the race's entrants are all finished or forfeited, you can only chat if the difference between the current time and the most recent finished_at/forfeited_at time <= X?

Reply:
Yep, but you should probably take a look at [this](https://github.com/glacials/splits-io/pull/550#discussion_r283105684) reply in regards to figuring out if the race is finished.